### PR TITLE
feat: create versioned purls from OSV

### PR DESCRIPTION
**Pull request by ctron**
Also see: https://github.com/trustification/trustify/issues/870

### CODE REVIEW

File: modules/ingestor/src/service/advisory/osv/loader.rs [hunk 1]
```diff
@@ -104,97 +107,85 @@ impl<'g> OsvLoader<'g> {
...
+                        }
+
+                        if let (_, Some(fixed)) = &parsed_range {
+                            advisory_vuln
+                                .ingest_package_status(
+                                    None,
+                                    &purl,
+                                    "fixed",
+                                    VersionInfo {
+                                        // TODO detect better version scheme
```
Comment (JimFuller-RedHat) [inline modules/ingestor/src/service/advisory/osv/loader.rs:174]: can we expand on this or raise an issue with what you are thinking ?

File: modules/ingestor/src/service/advisory/osv/loader.rs [hunk 2]
```diff
@@ -104,97 +107,85 @@ impl<'g> OsvLoader<'g> {
...
+
+                let Some(package) = &affected.package else {
+                    continue;
+                };
 
-                    purls.extend(translate::to_purl(package).map(Purl::from));
+                // extract PURLs
+
+                let mut purls = vec![];
+                purls.extend(translate::to_purl(package).map(Purl::from));
```
Comment (JimFuller-RedHat) [inline modules/ingestor/src/service/advisory/osv/loader.rs:119]: is this some kind of hyper specific translate to purl ?

File: modules/ingestor/src/service/advisory/osv/loader.rs [hunk 3]
```diff
@@ -104,97 +107,85 @@ impl<'g> OsvLoader<'g> {
             }
 
             for affected in &osv.affected {
-                if let Some(package) = &affected.package {
-                    let mut purls = vec![];
+                // we only process it when we have a package
```
Comment (JimFuller-RedHat) [inline modules/ingestor/src/service/advisory/osv/loader.rs:110]: might be worth a log::trace!

**Review by JimFuller-RedHat — COMMENTED** [review]
I do not think existing test has any assertions for checking versions for this specific fix might be worth adding, otherwise minor comments - LGTM

File: modules/ingestor/src/service/advisory/osv/loader.rs (hunk 2 above)
Comment (ctron) [reply modules/ingestor/src/service/advisory/osv/loader.rs:119]: Yup.

**Pull request closed without merging**